tests/model_serving/model_runtime/vllm/utils.py

```
@@ -199,3 +206,87 @@ def validate_raw_tgis_inference_request(
 def skip_if_not_deployment_mode(isvc: InferenceService, deployment_type: str) -> None:
     if isvc.instance.metadata.annotations["serving.kserve.io/deploymentMode"] != deployment_type:
         pytest.skip(f"Test is being skipped because model is not being deployed in {deployment_type} mode")
+
+
+#Performance Helper Methods
+
+def get_vllm_version(namespace, pod_name):
+    cmd = f'oc exec -n {namespace} {pod_name} -- python -c "import vllm; print(vllm.__version__)"'
+    result = subprocess.check_output(cmd, shell=True, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)